src/components/SearchBar.jsx

```diff
@@ -1,64 +1,94 @@
-import { useEffect, useState } from "react"
-import { getArticles, getTopics } from "../api"
-import Loading from "./Loading"
+import { useEffect, useState } from 'react';
+import { getArticles, getTopics } from '../api';
+import Loading from './Loading';
 
-function SearchBar({setArticlesList}) {
- 
-  const [topicsList, setTopicsList] = useState()
-  const [topicQuery, setTopicQuery] = useState()
-  const [isLoading, setIsLoading] = useState(true)
-	const [error, setError] = useState(false)
+function SearchBar({ setArticlesList }) {
+	const [topicsList, setTopicsList] = useState();
+	const [topicQuery, setTopicQuery] = useState();
+	const [sortQuery, setSortQuery] = useState();
+	const [orderQuery, setOrderQuery] = useState();
+	const [isLoading, setIsLoading] = useState(true);
+	const [error, setError] = useState(false);
 
-  useEffect(() => {
+	useEffect(() => {
     setError(false)
-    getTopics().then((topics) => {
-      setTopicsList([{slug: '' }, ...topics])
-      setIsLoading(false)
-    }).catch((err) => {
-      setError(true);
-      setIsLoading(false);
-    });
-  }, [topicsList])
+		getTopics()
+			.then((topics) => {
+				setTopicsList([{ slug: '' }, ...topics]);
+				setIsLoading(false);
+			})
+			.catch((err) => {
+				setError(err);
+				setIsLoading(false);
+			});
+	}, [topicsList]);
 
-  const handleChange = (e) => {
-    setTopicQuery(e.target.value)
-  } 
+	const handleChangeTopic = (e) => {
+		setTopicQuery(e.target.value);
+	};
 
-  const handleClick = (e) => {
-    e.preventDefault()
+	const handleChangeSort = (e) => {
+		setSortQuery(e.target.value);
+	};
 
-    getArticles(topicQuery).then(({articles}) => {
-      setArticlesList(articles)
-      setIsLoading(false)
-    }).catch((err) => {
-      setError(true)
-      setIsLoading(false)
-    })
-  }
+	const handleChangeOrder = (e) => {
+		setOrderQuery(e.target.value);
+	};
 
-  if (error) {
+	const handleClick = (e) => {
+		e.preventDefault();
+
+		getArticles(topicQuery, sortQuery, orderQuery).then(({ articles }) => {
+			setArticlesList(articles);
+			setIsLoading(false);
+		});
+	};
+
+	if (error) {
 		return (
 			<section className="errorMessage">
-				<p>{error.request.status}: {error.response.data.msg}</p>
+				<p>
+					{error.request.status}: {error.response.data.msg}
+				</p>
 			</section>
 		);
 	}
 
-  if(isLoading){
-    return <Loading />
-  } else {
-    return (
-      <section className="searchBar">
-         <label htmlFor="category">Topics</label>
-        <select onChange={handleChange} id='category'type="text">
-         { topicsList.map((topic) => {
-          return  <option key={topic.slug} value={topic.slug}>{topic.slug}</option>
-         })}
-        </select>
-         <button onClick={handleClick} className="btn">Send</button>
-      </section>
-    )
-  }
+	if (isLoading) {
+		return <Loading />;
+	}
+
+	return (
+		<section className="searchBar">
```

Comment:
Great stuff on this input process, we are most of the way there to implementing this as a form. Consider how we can finish this off nicely to ensure it is fully controlled by react. The nc notes on controlled components may be helpful to refer back to here :)

Hint: Think of this as a form with an onSubmit 👍